### PR TITLE
RavenDB-7070 in the tests use AsyncManualResetEvent instead of ManualResetEvent/Slim 

### PR DESCRIPTION
**Pull request by karmeli87**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-7070

### CODE REVIEW

**Comment by arekpalinski** [comment]
`SlowTests.Issues.RavenDB_17231.FailuresDuringEmptyQueueShouldCauseReelection` has failed each time:

https://jenkins.hibernatingrhinos.com/job/PR_Tests/10289/
https://jenkins.hibernatingrhinos.com/job/PR_Tests/10285/
https://jenkins.hibernatingrhinos.com/job/PR_Tests/10286/

`SlowTests.Issues.RavenDB_19548.RawRavenQueryProjectionWithoutNewKeyword` failed twice (on the builds running on ubuntu)

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 1]
```diff
@@ -30,8 +30,8 @@ public async Task RawRavenQueryProjectionWithoutNewKeyword()
                 }
 
                 /* should work:
-                * simple linq select
-                */
+                 * simple linq select
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB_19548.cs]: I don't think you should do any formatting changes in this file

File: test/SlowTests/Issues/RavenDB_8847.cs [hunk 2]
```diff
@@ -98,7 +98,7 @@ public void Should_set_first_batch_timeout_of_newly_created_static_index()
 
                     session.Query<User>(usersByname).Customize(x => x.WaitForNonStaleResults()).Where(x => x.Name != "ema").ToList();
 
-                    Indexes.WaitForIndexBatchCompleted(store, x => x.DidWork).Wait(TimeSpan.FromSeconds(2));
+                    Indexes.WaitForIndexBatchCompleted(store, x => x.DidWork).WaitAsync(TimeSpan.FromSeconds(2));
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB_8847.cs]: Missing `await`?

File: test/SlowTests/Issues/RavenDB_8847.cs [hunk 3]
```diff
@@ -48,7 +48,7 @@ public void Should_set_first_batch_timeout_of_newly_created_auto_index()
 
                     session.Query<User>().Customize(x => x.WaitForNonStaleResults()).Where(x => x.Name != "ema").ToList();
 
-                    Indexes.WaitForIndexBatchCompleted(store, x => x.DidWork).Wait(TimeSpan.FromSeconds(2));
+                    Indexes.WaitForIndexBatchCompleted(store, x => x.DidWork).WaitAsync(TimeSpan.FromSeconds(2));
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB_8847.cs]: Missing `await`?

File: test/Tests.Infrastructure/RavenTestBase.Indexes.cs [hunk 4]
```diff
@@ -181,11 +182,11 @@ public long WaitForEntriesCount(IDocumentStore store, string indexName, int minE
             return entriesCount;
         }
 
-        public ManualResetEventSlim WaitForIndexBatchCompleted(IDocumentStore store, Func<(string IndexName, bool DidWork), bool> predicate)
+        public AsyncManualResetEvent WaitForIndexBatchCompleted(IDocumentStore store, Func<(string IndexName, bool DidWork), bool> predicate)
```
Comment (arekpalinski) [inline test/Tests.Infrastructure/RavenTestBase.Indexes.cs]: Please double check all callers of this method

Comment (karmeli87) [reply test/Tests.Infrastructure/RavenTestBase.Indexes.cs]: yes, I reverted this method.

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 1 above)
Comment (karmeli87) [reply test/SlowTests/Issues/RavenDB_19548.cs]: yes, I reverted the formatting of this file

**Pull request merged**